### PR TITLE
Fix conflict between formatter and rules: spaces before new lines

### PR DESCRIPTION
**Pull request by PalumboN**
After discussion in https://github.com/pharo-project/pharo/pull/12387 between @jordanmontt and @astares we change the formatter to not enter in collision with the rule.

Beyond the rules that any of us want in his code, I think that is important that the tools inside the environment won't conflict between them (ie. the formatter should not break other systems like linter, compiler, highlighter, etc ). 
Should be nice to maintain this equilibrium, if you change something in the system keep in mind that should be in harmony with the rest of the components.

> PR made in a Pharo Sprint with @hernanmd

### CODE REVIEW

**Comment by Ducasse** [comment]
This is cool. 
As a default for EF I would like to have no extra line between the selector and the first comment - it is on my todo since ages.

**Comment by astares** [comment]
Nice as the formatter is the right place to take action.

Will try to check it but I'm busy with
https://github.com/pharo-project/pharo/issues/12431
which should also help to improve the situation. Willl work on that first.

**Comment by guillep** [comment]
@astares I'd like if you can make a review when you have time :)

**Comment by astares** [comment]
Will do!

File: src/EnlumineurFormatter-Tests/EFBlockExpressionTest.class.st [hunk 1]
```diff
@@ -124,6 +124,18 @@ EFBlockExpressionTest >> dontFormatCommentWithStatementConfiguration [
 		indentExtraSpaces:0
 ]
 
+{ #category : #configurations }
+EFBlockExpressionTest >> example: col [
```
Comment (jordanmontt) [inline src/EnlumineurFormatter-Tests/EFBlockExpressionTest.class.st]: What is this? It looks like some leftover

**Review by jordanmontt — APPROVED** [review]

**Comment by jordanmontt** [comment]
Clever solution, each time that there is a new line the spaces at the end of the statement are removed

**Comment by astares** [comment]
looks good from cases I tried to format - but I havent checked very deeply (not yet had the time to check code)

**Comment by PalumboN** [comment]
Thanks for the review!

@Ducasse I think that you can configure that with this setting:
image

**Comment by Ducasse** [comment]
Yes I did it but I was stupid not to put it on by default :)

**Review by astares — APPROVED** [review]

**Comment by guillep** [comment]
Thanks!

**Pull request merged**